### DISCUSSION TITLE
Remove diamonds rain animation

### DISCUSSION THREAD

**Discussion by brootle** [deso-protocol/dips#91] [Medium CIPs]
Please, remove "**_diamonds rain_**" animation. It was fun at the beginning, but it creates terrible user experience. Under certain conditions it makes browser unresponsive. Thank you!

To reproduce, go to https://bitclout.com/u/brootle and try to diamond every post on the wall, after 20-30 clicks on diamond in a row it becomes really slow, after 50 clicks it's basically dead, I am testing from different browsers and different accounts, to do 100 clicks on a diamond in a row is practically impossible.

**Comment by bloated-devish** [comment]
Should we reduce the number of diamonds? Do you want to run a test where you reduce the number of "confetti" items or clock to see whether that makes it better? Better produce a chart to show data on what sounds better to you.

Is this the place to change the duration diamonds fly? You can also fiddle with max.
https://github.com/bitclout/frontend/blob/main/src/app/global-vars.service.ts#L639

**Reply by brootle** [reply]
just edited post describing how I am testing it, looks like diamond shower is getting stuck in the memory depending on how many times you trigger it

**Reply by bloated-devish** [reply]
I wonder if this is defective by design - to ensure people don't manipulate giving diamonds bulk. hahaha

**Comment by tijno** [comment]
Im in favour of removing it fully.

**Comment by seanslater** [comment]
I'm going to go against the tide here and say that the diamond animation needs to say - the giving of diamonds is on of the unique parts of BitClout and the animation you receive when you give them is a little dopamine hit.  While it may be getting a little tired for users that have been on the platform for awhile - its something that catches the eye of new users.

I would ask is anyone really clicking to diamond without a page reload 30 times?  Often enough to warrant getting rid of them?

Perhaps an option could be made in the settings panel 'Disable Animations' - allowing new users to experience it, but others to disable if they choose.

tldr; the animation is going to excite new users, and we want that, so I don't think the animation should be removed.

**Reply by iPaulPro** [reply]
Agreed. Settings toggle is the best option.

**Reply by matt-mcmarsh** [reply]
agree a toggle in settings if possible would be best
new users may appreciate the confetti
and then we have the option to speed things up and turn it off if we want :-)

**Reply by tijno** [reply]
toggle is a good idea - but rather then it being "hide diamonds" - maybe it should be something like "reduce motion & animations"

**Comment by ItsAditya-xyz** [comment]
Oh yes! Please do this for the sake of UI. I know art is good but talk about design please. If diamond rain will be removed, I believe people are more likely to give more diamonds. Currently giving diamonds isn't as smooth as liking a post.
